cpp/src/arrow/compute/kernels/vector_replace.cc

```diff
@@ -119,10 +132,10 @@ void ReplaceWithArrayMaskImpl(const ArrayData& array, const ArrayData& mask,
       // Copy from replacement array
       Functor::CopyData(*array.type, out_values, out_offset + write_offset, replacements,
                         replacements_offset, block.length);
-      if (replacements_bitmap) {
+      if (replacements_bitmap && Functor::HasBitmap(replacements)) {
```

Comment:
`replacements_bitmap` is already a test for whether we have a validity bitmap in the replacements array - why another check?